File: Arrays & Basic Complexity/sol.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> res;
        map<int, int> mp;
        for(int i = 0; i< nums.size(); i++){
            if(mp.find(target - nums[i]) == mp.end()){
                mp[nums[i]] = i;
            }
            else{
                res.push_back(mp[target - nums[i]]);
                res.push_back(i);
                return res;
            }
        }
        return res;
    }
};

File: Arrays & Basic Complexity/sol.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let map = {};
    for (let i = 0;i<nums.length;i++) {
        let tamamlayici = target - nums[i]
        if (map.hasOwnProperty(tamamlayici)) {
            return [map[tamamlayici],i];
        }
        map[nums[i]] = i;
    }
};